### DISCUSSION TITLE
Purescript codegen plugin

### DISCUSSION THREAD

**Discussion by roryc89** [dotansimha/graphql-code-generator#6230] [Ideas]
We have built a [Purescript graphql client](https://github.com/OxfordAbstracts/purescript-graphql-client) that already includes type codegen.

At the moment we have site that demos the codegen but it would seem reasonable to include it here as well. I can make a PR to include it a plugin here if it would be good fit for this project.

**Comment by dotansimha** [comment]
Hi @roryc89 ! Thank you for creating this issue. 
This is great to see the GraphQL community evolving in other ecosystems. 
I took a quick look and it seems like the codegen you have there is not built on top of GraphQL Codegen? Seems like it's a standalone tool? 

I think the effort of using the GraphQL Codegen infrastructure and the wrap the existing code and GraphQL Codegen plugin should be minimal, and you can leverage that in order to remove the need for loading/downloading the schemas, since Codegen can do that for you.

> Also, moving to Discussions.
